plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1298,8 +1302,9 @@ public Optional<ConnectorOutputMetadata> finishInsert(
             // it is not obvious why we need to persist this readVersion
             transactionLogWriter.appendCommitInfoEntry(getCommitInfoEntry(session, commitVersion, createdTime, INSERT_OPERATION, handle.getReadVersion()));
 
-            // Note: during writes we want to preserve original case of partition columns
-            List<String> partitionColumns = handle.getMetadataEntry().getOriginalPartitionColumns();
+            List<String> partitionColumns = getColumnMappingMode(handle.getMetadataEntry()) == ColumnMappingMode.NAME
```

Comment:
shouldn't we apply the same handling in `finishCreateTable` and `finishOptimize` as well?

Also shouldn't `io.trino.plugin.deltalake.DeltaLakeOutputTableHandle#getPartitionedBy` use `getPhysicalName()` instead of `getName()` ?

https://github.com/trinodb/trino/blob/c98bb407ea31b858e6c556dc39c6a790f50e6120/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeOutputTableHandle.java#L91-L97

Reply:
can you pls add tests for these two situations in `TestDeltaLakeColumnMappingMode`?

Reply:
Nevermind, these changes are not yet in the scope of this PR.

```
            assertQueryFailure(() -> onTrino().executeQuery("ALTER TABLE delta.default." + tableName + " EXECUTE OPTIMIZE"))
                    .hasMessageContaining("Delta Lake writer version 5 which is not supported");
```